dbms/src/Common/Spiller.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Core/Block.h>
+#include <DataStreams/IProfilingBlockInputStream.h>
+#include <DataStreams/NativeBlockInputStream.h>
+#include <IO/CompressedReadBuffer.h>
+#include <IO/ReadBufferFromFile.h>
+#include <Poco/File.h>
+
+namespace DB
+{
+class SpilledFile : public Poco::File
+{
+public:
+    explicit SpilledFile(const String & file_name);
+    ~SpilledFile() override;
+    void addSpilledDataSize(size_t added_size) { spilled_data_size += added_size; }
+    size_t getSpilledDataSize() const { return spilled_data_size; }
+
+private:
+    size_t spilled_data_size = 0;
+};
+
+struct SpilledFileStream
+{
+    ReadBufferFromFile file_in;
+    CompressedReadBuffer<> compressed_in;
+    BlockInputStreamPtr block_in;
+
+    explicit SpilledFileStream(const std::string & path, const Block & header)
+        : file_in(path)
+        , compressed_in(file_in)
+        , block_in(std::make_shared<NativeBlockInputStream>(compressed_in, header, 0))
+    {}
+};
+
+class SpilledFilesInputStream : public IProfilingBlockInputStream
```

Comment:
Just a suggestion, make the class as a separate file.

Reply:
Move `SpilledFilesInputStream` to a seperate file.